core/domain/platform_parameter_domain.py

```diff
@@ -40,8 +40,10 @@
     FEATURE_STAGES.dev, FEATURE_STAGES.test, FEATURE_STAGES.prod]
 ALLOWED_CLIENT_TYPES = ['Web', 'Android']
 ALLOWED_BROWSER_TYPES = ['Chrome', 'Edge', 'Safari', 'Firefox', 'Others']
+ALLOWED_APP_VERSION_FLAVOR = ['alpha', 'beta', 'test', 'n/a']
 
-APP_VERSION_WITH_HASH_REGEXP = re.compile(r'^(\d+(?:\.\d+)*)(?:-[a-z0-9]+)?$')
+APP_VERSION_WITH_HASH_REGEXP = re.compile(
```

Comment:
Actually, I think I'm changing my perspective on this slightly. Let's handle the case when the flavor is missing & default to unknown, but let's also handle & expect that release is present. E.g. "1.2.3-abcdef-release" is a valid version that we likely will end up using. We should make the system robust for cases when the flavor is incorrectly omitted, though. If it's missing, no filters will end up matching since it's in an undesired state.

Reply:
Done, added support for 'release' flavor & changed 'n/a' to 'unknown'.